comment_scan/comment_scanner.rb

```diff
@@ -21,13 +24,38 @@ def scan_new_comments
         new_comments = @seclient.comments_after_date(@latest_comment_date)
         @latest_comment_date = new_comments[0].json["creation_date"].to_i+1 if new_comments.any? && !new_comments[0].nil?
         scan_se_comments([new_comments])
+        check_for_hot_post([new_comments])
     end
 
     def tick
         (@time_to_scan = 60; return false) if (@time_to_scan -= 1) <= 0
         return true
     end
 
+    def check_for_hot_post(new_comments)
+        #Only check each post once, so make unique by post ID
+        new_comments.flatten.uniq(&:post_id).each do |comment|
+            continue unless post = @seclient.post_exists?(comment.post_id) # If post was deleted, skip it
+            comments_on_post = Comment.
+                                    where(post_id: comment.post_id).
+                                    where("creation_date >= :date", date: Time.at(comment.creation_date - @hot_seconds).to_datetime)
+
+            if comments_on_post.count >= @hot_comment_num && !MessageCollection::ALL_ROOMS.hot_post_recorded?(post.id)
+                report_hot_post(post.link, post.title, comments_on_post.count, @hot_seconds/60/60)
+                MessageCollection::ALL_ROOMS.push_hot_post(post.id)
+            end
+        end
+    end
+
+    def report_hot_post(post_link, post_title, comment_num, hr_num)
+        (@chatter.rooms + [@chatter.HQroom]).flatten.each do |room_id|
```

Comment:
Can't this be `@chatter.rooms.push(@chatter.HQroom).each`?

Reply:
Wouldn't that then add HQ to the rooms list? The goal here is to keep it off that list to make everything we do to only child rooms easier.

Reply:
Yep, that was dumb of me. But still, you shouldn't need to flatten this, right? `@chatter.room` is just a 1d array, right?

Reply:
Huh. Good question. Things definitely work just fine without it. Let me fix that...